New Template files for Cassandra Connector

mcherem opened the pull request:
Fit files to new template.

mcherem committed:
Fit files to template

Fit files to template

mcherem committed:
Update cassandra-config-topics.adoc

Fixed title.

mcherem committed:
Update cassandra-design-center.adoc

Fixed title.

mcherem committed:
Update cassandra-get-started.adoc

Fixed title & formatting issues.

mcherem committed:
Update cassandra-studio.adoc

Fix title.

mcherem committed:
Update cassandra-use-cases.adoc

Fix title.

mcherem committed:
Update cassandra-xml-maven.adoc

Fixed title.

mcherem committed:
Update cassandra-config-topics.adoc

Removed irrelevant heading per George.

mcherem committed:
Update cassandra-design-center.adoc

Added a first step per George.

mcherem committed:
Update cassandra-get-started.adoc

Added Cassandra to Connector

mcherem committed:
Update cassandra-design-center.adoc

REmoved the "To"

mcherem committed:
Update cassandra-get-started.adoc

Updated title.

george-hoffer commented on modules/ROOT/pages/cassandra/cassandra-get-started.adoc:
```diff
@@ -0,0 +1,43 @@
+
+The Cassandra connector implements the DataStax Java Driver for Apache Cassandra 3.1. The connector exposes multiple operations in the Cassandra Connector Reference.
+
+Certain operations accept some custom POJOs as input parameters, and these are needed for the DataSense functionality.
```
Please move this paragraph and the next to the Prerequisites section -- they're too implementation-oriented for a starting intro.

mcherem replied:
Done.

george-hoffer commented on modules/ROOT/pages/cassandra/cassandra-get-started.adoc:
```diff
@@ -0,0 +1,43 @@
+<No existing examples>
+
+== Triggers
+. In Design Center, click Set Up > Upload, browse for and select the driver for this connector on your file system, and upload it. Or, search for and select a driver that is already uploaded.
```
This is incorrect -- A trigger is only a way to start a Mule flow with the current connector

mcherem replied:
Will update Trigger description and move to Design & Studio sections.

george-hoffer commented on modules/ROOT/pages/cassandra/cassandra-studio.adoc:
```diff
@@ -0,0 +1,111 @@
+= Cassandra-Connector.adoc - Studio - Mule 4
```
Change the title to Cassandra - Studio - Mule 4

mcherem replied:
Fixed.

george-hoffer commented on modules/ROOT/pages/cassandra/cassandra-studio.adoc:
```diff
@@ -0,0 +1,111 @@
+= Cassandra-Connector.adoc - Studio - Mule 4
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
```
Should add ":imagesdir: ../../assets/images/" immediately after endif::[] so you can see images in Git

mcherem replied:
Oops, got carried away and took this out. It's back now.

george-hoffer commented on modules/ROOT/pages/cassandra/cassandra-use-cases.adoc:
```diff
@@ -0,0 +1,232 @@
+include::_attributes.adoc[]
+endif::[]
+
+== Use Case: Studio
```
Say what the Use Case does -- Use Case: Studio won't help customers understand why to read this

mcherem replied:
Gave it a name.

george-hoffer commented on modules/ROOT/pages/cassandra/cassandra-use-cases.adoc:
```diff
@@ -0,0 +1,232 @@
++
+. Drag an HTTP connector onto the canvas and leave the default values for Host and Port and set the path to `/test/createKeyspace`.
++
+.Drag a Transform Message component from the Mule palette to the right (process) side of the flow, select the component, and set the output payload. For example:
```
Add a space between the dot and "Drag"

mcherem replied:
Fixed.

george-hoffer commented on modules/ROOT/pages/cassandra/cassandra-xml-maven.adoc:
```diff
@@ -0,0 +1,56 @@
+endif::[]
+
+== Add a Namespace for the Connector
+<No Namespace>
```
Namespace:

```
[source,xml,linenums]
----
http://www.mulesoft.org/schema/mule/cassandra-db
http://www.mulesoft.org/schema/mule/cassandra-db/current/mule-cassandra-db.xsd
----
```

mcherem replied:
Fixed.

george-hoffer commented on modules/ROOT/pages/cassandra/cassandra-xml-maven.adoc:
```diff
@@ -0,0 +1,56 @@
+
+
+----
+<No Dependency Snippets>
```
remove this source block since you list the pom file info already

mcherem replied:
Fixed.

mcherem committed:
Update cassandra-get-started.adoc

Update template changes from 5/2 & George's feedback.

mcherem committed:
Update cassandra-studio.adoc

RE added: :imagesdir: ../../assets/images/

mcherem committed:
Update cassandra-use-cases.adoc

Edits from George.

mcherem committed:
Update cassandra-xml-maven.adoc

George's edits.

mcherem committed:
Update cassandra-config-topics.adoc

File edits.

mcherem committed:
Update cassandra-use-cases.adoc

Edits.

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-config-topics.adoc:
```diff
@@ -0,0 +1,27 @@
+include::_attributes.adoc[]
+endif::[]
+
+The Cassandra DB connector configuration must match the Cassandra configuration with regard to Basic Authentication.
```
You Initial cap Cassandra Connector later on. We need to add some rules for when to initial cap Connector and when not to,

mcherem replied:
Were you able to check the Salesforce Style Guide for rules regarding this item? If it wasn't there, do you want to start a list of questions?

mcherem replied:
Fixed to be consistent.

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-config-topics.adoc:
```diff
@@ -0,0 +1,27 @@
+
+The Cassandra DB connector configuration must match the Cassandra configuration with regard to Basic Authentication.
+By default, accessing the Cassandra database does not require credentials, so you need to configure Cassandra to enable authentication.
+After enabling authentication, run the `cqlsh` command from the Cassandra installation home directory as shown in this example. It uses default credentials with the username cassandra and password cassandra:
```
Should the username and password be bolded?

mcherem replied:
I don't know the rules around this. What does the Salesforce style guide say?

mcherem replied:
Will bold unless I hear otherwise.

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-design-center.adoc:
```diff
@@ -0,0 +1,118 @@
+endif::[]
+
+== Configure the Cassandra Connector in Design Center
+. From Anypoint Platform > Design Center, create a Mule Application, and add an HTTP Listener as the trigger.
```
I think we are using the term app now and not application, but this may be different for the connector doc. Application should be lowercase.

mcherem replied:
Don't know, can you find out for sure?

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-design-center.adoc:
```diff
@@ -0,0 +1,118 @@
+|Operation |Description
+|Add New Column a|Input for this operation is:
+
+* Parameter: A table name
```
Either have the parameter descriptions start with A or not. Right now, the text is inconsistent. The capitalization of the A and an at the beginning of the descriptions are also inconsistent.

mcherem replied:
Okay, duly noted.

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-design-center.adoc:
```diff
@@ -0,0 +1,118 @@
++
+The POJO has the following fields:
++
+** Name of the column to be created
```
Same comment about using or not using "the" at the start of the description. I won't mark these anymore.

mcherem replied:
Noted, will fix in all.

mcherem replied:
Fixed.

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-design-center.adoc:
```diff
@@ -0,0 +1,118 @@
+
+* A table name
+* The keyspace that contains the table
+|Execute CQL Query  a| Transform Message Payload.
```
I don't think payload should be initial capped,

mcherem replied:
Will need to double check on this one.

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-design-center.adoc:
```diff
@@ -0,0 +1,118 @@
+[%header%autowidth.spread]
+|===
+|Field |Description
+|Host | Enter a host name, or IP address, of a Cassandra node.
```
Remove the comma after "host name." Actually, I would say: "Enter the host name or IP address of a Cassandra node."

mcherem replied:
Yes.

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-get-started.adoc:
```diff
@@ -0,0 +1,37 @@
+
+Cassandra Connector v1.0
+
+The Anypoint CassandraDB connector creates connectivity between Cassandra databases and third-party applications, systems, services such as SAP, PeopleSoft, or Workday as well as other databases across the enterprise. Apache Cassandra is an open source distributed NoSQL database intended to manage large amounts of product, customer, and employee data. 
```
Add a comma after Workday,

mcherem replied:
Don't agree with this one. STET

Mulesoft-Mysti replied:
Salesforce style is serial comma.

mcherem replied:
Fixed.

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-get-started.adoc:
```diff
@@ -0,0 +1,37 @@
+Certain operations accept some custom POJOs as input parameters, and these are needed for the DataSense functionality.
+These are: addNewColumn, changeColumnType, createKeyspace, createTable, deleteColumnsValue, deleteRows, dropColumn, dropKeyspace, dropTable, executeCQLQuery, getTableNamesFromKeyspace, renameColumn, select, and update.
+
+Familiarity with Cassandra, Mule, Anypoint Connectors, Anypoint Studio, Mule concepts, elements in a Mule flow, and Global Elements.
```
I am not sure whether Global Elements should be initial capped.

mcherem replied:
What are you basing your question on? What do you think it means?

Mulesoft-Mysti replied:
Global Elements is the UI label. I'd call them Mule global elements (just so the customer, if unfamiliar, has some idea what we're talking about). Also, telling the customer they must be familiar with "Mule concepts" is not helpful, I'd make it more specific or delete it, especially since we then call out two Mule concepts, elements in a flow and global elements. Finally, to avoid confusing customers, can we just say "Mule flows" to not say "elements" twice so close together? 

Something like: Be familiar with basic concepts:
* Cassandra (what? data flow? any way to be more specific?)
* Anypoint Connectors (they're getting started. What are they supposed to be familiar with?)
* Mule runtime engine, especially Mule flows and global elements
* The environment you use to work with connectors, either Anypoint Studio or Design Center (? isn't there a 3rd audience?)

mcherem replied:
Again, this is generic template wording that I hope Diane and Mysti will forward to Meera.

mcherem replied:
Template feedback to be resolved with Meera.

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-use-cases.adoc:
```diff
@@ -0,0 +1,231 @@
+|Password | Enter the password if you enabled PasswordAuthenticator. Otherwise leave this value blank.
+|===
++
+. Click the Test Connection option to confirm that Mule can connect with the Cassandra instance. If the connection is successful, click OK to save the configuration. Otherwise, review and correct any invalid parameters and retest.
```
Should OK be bolded?

mcherem replied:
Don't know, what does the Salesforce style guide say?

Mulesoft-Mysti replied:
yes, all UI controls are bold.

mcherem replied:
Fixed.

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-use-cases.adoc:
```diff
@@ -0,0 +1,231 @@
+config.password=<PASSWORD>
+----
++
+. Drag an HTTP connector onto the canvas and leave the default values for Host and Port and set the path to `/test/createTable`.
```
Remove the first and, and use a comma instead.

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-use-cases.adoc:
```diff
@@ -0,0 +1,231 @@
++
+. Click the Test Connection option to confirm that Mule can connect with the Cassandra instance. If the client test is successful, click OK to save the configuration. Otherwise, review or correct any invalid parameters and retest.
++
+. Run the Mule app. In Postman, select POST. Select Body > Raw, select the JSON (application/json) MIME type, and enter the following table description using uppercase for data types.
```
This is a George or Meera question, but in the Mulesoft Fundamentals Class, they are using the Advanced REST Client for walkthroughs. Postman is very popular though.

mcherem replied:
Can you make sure the team knows the answer to this once you find out?

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-get-started.adoc:
```diff
@@ -0,0 +1,37 @@
+
+ * New user -- Read this document and configure your Mule application in xref:cassandra/cassandra-design-center.adoc[Design Center].   
+* Current Mule user -- Read this document and configure your Mule application in xref:cassandra/cassandra-design-center.adoc[Design Center] or xref:cassandra/cassandra-studio.adoc[Anypoint Studio]. 
+The xref:cassandra/cassandra-config-topics.adoc[Configuration Topics] help you extend your understanding of Salesforce actions.
```
It is not clear what "Salesforce acctions" are.

Mulesoft-Mysti replied:
We probably want to be a bit more specific here, there are a lot of SFDC actions: https://developer.salesforce.com/docs/atlas.en-us.api_action.meta/api_action/actions_intro_list.htm#

Why is line 16 under Audience and only for current Mule user? What is this section for? Are we saying only these three types of people should expect to use Cassandra connector? Are we claiming the listed docs are the only ones needed for user of type listed? What problem is this section solving?

mcherem replied:
This is template feedback which I think Diane and Mysti will forward to Meera.

mcherem replied:
All good generic template questions to pass along to Meera.

dhirsch1 commented on modules/ROOT/pages/cassandra/cassandra-get-started.adoc:
```diff
@@ -0,0 +1,37 @@
+
+The Cassandra connector implements the DataStax Java Driver for Apache Cassandra 3.1. The connector exposes multiple operations in the Cassandra Connector Reference.
+
+Certain operations accept some custom POJOs as input parameters, and these are needed for the DataSense functionality.
```
I suggest:

The following operations accept some POJOs as input parameters that are needed for the DataSense functionality:

Mulesoft-Mysti replied:
It'd be great if this could be worded in less passive construction. But I don't understand what the point is. Are we telling people if they use DataSense that custom POJOs are needed, and only some operations support that (if so, dhirsch1 rewrite provides the right emphasis)? Why are we saying this here in the getting started? Why don't we say it in the operation description?

mcherem replied:
I don't know for sure, but trust Diane's rewrite as she seems to have a better understanding of the issue here.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-get-started.adoc:
```diff
@@ -32,5 +38,6 @@ If a global element is not required, use an HTTP Listener or a Scheduler as the
 +
 
 == See Also
+
```
I know Salesforce has a rule in their style guide about having only 3 links in See Also, but it seems like a helpful thing to put the pre-req doc links here as well. ???

mcherem replied:
We are supposed to resolve this with Meera. I agree that maybe it's worth deviating if the info is helpful.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-get-started.adoc:
```diff
@@ -1,12 +1,16 @@
-= Cassandra-Connector.adoc Connector - Get Started - Mule 4
+= Get Started - Mule 4
```
The order of sections in this topic confuses me (I think this comes from the template)? And this doesn't feel like a getting started topic because there are no tasks.

Mulesoft-Mysti replied:
Why do we specify Mule 4 and not Cassandra Connector in the title?

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-studio.adoc:
```diff
@@ -0,0 +1,112 @@
+= Studio - Mule 4
```
I don't think this is a best practice as a topic title since it leaves out Cassandra. In google searches, would this show up?

mcherem replied:
Good point, hope this will get passed along to Meera.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-studio.adoc:
```diff
@@ -0,0 +1,112 @@
+= Studio - Mule 4
```
What is this topic for? task, reference, concept? And why does it mention Mule 4 but not Cassandra?

mcherem replied:
It's supposed to be a Studio How to. Again, hoping  this goes up to Meera.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-studio.adoc:
```diff
@@ -0,0 +1,112 @@
+|Operation |Description
+|Add New Column a|Input for this operation is:
+
+* Parameter: A table name
```
Could description be more useful, like "The table containing data you need to integrate" ?

mcherem replied:
Yes, but I'd like to know more about this before I change the heading. Will pass on it for this review.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-studio.adoc:
```diff
@@ -0,0 +1,112 @@
+|Add New Column a|Input for this operation is:
+
+* Parameter: A table name
+* Optional keyspace name
```
How does customer know what a keyspace name is or where to obtain the value? A hint here would be helpful.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-studio.adoc:
```diff
@@ -0,0 +1,112 @@
+
+* Parameter: A table name
+* Optional keyspace name
+* Payload: an AlterColumnInput POJO
```
Where does the POJO come from? How does the customer find the name (or is the name always AlterColumnInput?)

mcherem replied:
All good questions I will ask when I delve a little deeper into the connector.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-studio.adoc:
```diff
@@ -0,0 +1,112 @@
+* Optional keyspace name
+* Payload: an AlterColumnInput POJO
++
+The POJO has the following fields:
```
This can just be POJO fields:

mcherem replied:
Fixed.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-studio.adoc:
```diff
@@ -0,0 +1,112 @@
++
+The POJO has the following fields:
++
+** Name of the column to be created
```
Do the fields not have names & values?

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-studio.adoc:
```diff
@@ -0,0 +1,112 @@
+|Create Keyspace a| Input for this operation using SimpleStrategy is:
+
+* Keyspace name
+* Replication factor
```
What's this? Where does the customer find it?

mcherem replied:
Good question, I will have to investigate this on my second pass through the connector after putting them all in the templates.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-use-cases.adoc:
```diff
@@ -0,0 +1,231 @@
+= Use Cases - Mule 4
```
Again--these are not titles containing text that a customer would search for in Google.

Mulesoft-Mysti replied:
If it's a file of examples, call them examples in the title.

mcherem replied:
Hopefully you got Meera to change this.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-use-cases.adoc:
```diff
@@ -0,0 +1,231 @@
+. In Postman, click Send, and look for Status: 200 OK.
+
+
+== XML Flows
```
I should have some sense for what this section contains from the title, but XML Flows feels very out of context. The heading should probably be something more indicative, like "Try the Example" or something. This is a task section in a reference/conceptual topic. Also, *where* in Studio to I put it?

mcherem replied:
Fixed.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-xml-maven.adoc:
```diff
@@ -0,0 +1,59 @@
+= XML and Maven Support - Mule 4
```
Again, if we mention Mule 4, for SEO we should also have Cassandra in the title.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-xml-maven.adoc:
```diff
@@ -0,0 +1,59 @@
+include::_attributes.adoc[]
+endif::[]
+
+XML and POM file configuration information and XML-related usage topics.
```
This topic does not have enough context for me to understand. Is this all and only what I have to do? How is this different from the XML example I saw in Use Cases? Why didn't Use Cases mention maven/pom?

mcherem replied:
Feedback for Meera?

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-xml-maven.adoc:
```diff
@@ -0,0 +1,59 @@
+XML and POM file configuration information and XML-related usage topics.
+
+== Add a Namespace for the Connector
+
```
Where do I add this? I'm lost!

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-xml-maven.adoc:
```diff
@@ -0,0 +1,59 @@
+----
+
+== See Also
+* https://forums.mulesoft.com[MuleSoft Forum]
```
Instead of generic links, can we point people to MuleSoft doc that explains things? What is the relationship of this XML to a Mule app? What am I doing? Why?

mcherem replied:
Agreed, we should decide this for all templates.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-xml-maven.adoc:
```diff
@@ -0,0 +1,59 @@
+== Configure the Maven POM File
+When creating an app manually from the XML, set up the pom.xml for your project:
+
+ [source,xml,linenums]
```
This is rendering as part of the code--I think if you remove the space it'll work? Same for line 34.

Mulesoft-Mysti commented on modules/ROOT/pages/cassandra/cassandra-use-cases.adoc:
```diff
@@ -0,0 +1,231 @@
+
+== Example: Create a Cassandra Keyspace using Studio
+
+Create a Cassandra keyspace which groups column families, then create a Cassandra table.
```
If my goal is to use Cassandra connector to integrate a Cassandra database with some other data store--what part of that process is this activity?

mcherem replied:
Fixed, I think.

Mulesoft-Mysti reviewed (commented):
Some of the topic title conventions may cause SEO problems.
We need more context in a few places (commented on).
I feel like we're mixing concept, task, and reference in a confusing way. It's hard to tell without all the topics, so feel free to push back if you feel it makes sense.

mcherem committed:
Update cassandra-config-topics.adoc

Added initial cap.

mcherem committed:
Update cassandra-design-center.adoc

Peer edits.

mcherem committed:
Update cassandra-get-started.adoc

Peer reviews.

mcherem committed:
Update cassandra-use-cases.adoc

Peer reviews.

mcherem committed:
Update cassandra-studio.adoc

Peer reviews.

mcherem committed:
Update cassandra-use-cases.adoc

Peer edit.

mcherem committed:
Update cassandra-xml-maven.adoc

Peer edits.

mcherem committed:
Update cassandra-design-center.adoc

Team edits on this template. Changes need to be validated.

mcherem committed:
Update cassandra-get-started.adoc

Updates per team review. Still needs PM validation.

mcherem committed:
Update cassandra-design-center.adoc

Fixed the trigger statement.

mcherem committed:
Update cassandra-design-center.adoc

Word fix.

mcherem committed:
Update cassandra-design-center.adoc

Updates from Diane's notes.

mcherem committed:
Update cassandra-use-cases.adoc

Doc edits.

mcherem committed:
Update cassandra-xml-maven.adoc

Edit cleanup.

mcherem committed:
Update cassandra-xml-maven.adoc

Edits.

mcherem committed:
Update cassandra-xml-maven.adoc

Edits.

mcherem committed:
Update cassandra-xml-maven.adoc

Changed headings.

mcherem committed:
Update cassandra-xml-maven.adoc

REmoved heading edits.

mcherem committed:
Update cassandra-design-center.adoc

Added: After completing these steps, see the Examples section to experiment with the use cases described there.

mviswana reviewed (commented):
Test

mcherem committed:
Update cassandra-get-started.adoc

Updates to current template.

mcherem committed:
Update cassandra-get-started.adoc

Changes per template.

mviswana committed:
Update cassandra-get-started.adoc

Small tweaks to the different section. Removing template content for sections that don't have any information.

mcherem committed:
Update cassandra-get-started.adoc

Template updates

mcherem committed:
Update cassandra-design-center.adoc

Updated to latest templates.

mcherem committed:
Update cassandra-design-center.adoc

Fix numbering.

mcherem committed:
Update cassandra-studio.adoc

Template updates.

mcherem committed:
Update cassandra-config-topics.adoc

Template updates.

mcherem committed:
Update cassandra-xml-maven.adoc

Template updates.

mcherem committed:
Update cassandra-use-cases.adoc

Template updates.
